mteb/models/wav2vec2_models.py

```diff
@@ -0,0 +1,228 @@
+from __future__ import annotations
+
+from functools import partial
+from typing import Any, Iterable
+
+import numpy as np
+import torch
+import torchaudio
+from torch.utils.data import DataLoader
+from tqdm import tqdm
+from transformers import Wav2Vec2FeatureExtractor, Wav2Vec2ForCTC
+
+from mteb.encoder_interface import AudioBatch, AudioData, PromptType
+from mteb.model_meta import ModelMeta
+
+
+class Wav2Vec2AudioWrapper:
+    def __init__(
+        self,
+        model_name: str = "facebook/wav2vec2-xls-r-300m",
+        device: str = "cuda" if torch.cuda.is_available() else "cpu",
+        **kwargs: Any,
+    ):
+        self.model_name = model_name
+        self.device = device
+        self.model = Wav2Vec2ForCTC.from_pretrained(model_name).to(self.device)
+        self.feature_extractor = Wav2Vec2FeatureExtractor.from_pretrained(model_name)
+        self.sampling_rate = self.feature_extractor.sampling_rate
+
+    def _process_audio(self, audio: AudioBatch) -> list[torch.Tensor]:
+        processed_audio = []
+
+        if isinstance(audio, DataLoader):
+            for batch in audio:
+                processed_audio.extend(self._handle_batch(batch))
+        else:
+            processed_audio = self._handle_batch(audio)
+
+        return processed_audio
+
+    def _handle_batch(
+        self, batch: AudioData | Iterable[tuple[AudioData, str]]
+    ) -> list[torch.Tensor]:
+        waveforms = []
+
+        if isinstance(batch, tuple):  # Handle (audio, metadata) tuples
+            for audio, _ in batch:
+                waveforms.append(self._convert_audio(audio))
+        else:
+            for item in batch:
+                if isinstance(item, (np.ndarray, torch.Tensor)):
+                    waveforms.append(self._convert_audio(item))
+                elif isinstance(item, str):
+                    waveforms.append(self._load_audio_file(item))
+
+        return waveforms
+
+    def _convert_audio(self, audio: AudioData) -> torch.Tensor:
```

Comment:
Would this be more informative?
```suggestion
    def _convert_audio_from_numpy(self, audio: AudioData) -> torch.Tensor:
```